tasks/run_cd4pe_job.json

```diff
@@ -24,12 +24,17 @@
     },
     "docker_image": {
       "type": "Optional[String[1]]",
-      "description": "If specified, te job will attempt to run inside the docker container."
+      "description": "If specified, the job will attempt to run inside the docker container."
     },
     "docker_run_args": {
       "type": "Optional[Array[String[1]]]",
       "description": "If specified, the arguements will be passed to docker if docker image is specified."
     },
+    "docker_pull_creds": {
+      "type": "Optional[String[1]]",
+      "description": "Base64-encoded docker config.json to use when pulling the specified docker image.",
```

Comment:
Do user's have to provide the entire `config.json` or is there a way for them to just define credentials? It probably doesn't matter, I was just thinking of whether or not that would require them to set other values in the config file.

Reply:
The idea is that it's only used for the `docker pull`, and that CD4PE will provide those credentials (initially). This is modeled on the standard way to provide registry credentials in Kubernetes: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/

You generally provide a file that only supplies registry auth credentials.